tests/foreman/rhai/conftest.py

```diff
@@ -92,8 +95,64 @@ def test_foo(autosession):
             autosession.architecture.create({'name': 'bar'})
 
     """
-    test_name = '{}.{}'.format(request.module.__name__, request.node.name)
+    test_name = f'{request.module.__name__}.{request.node.name}'
     login = module_user.login
     password = module_user.password
     with Session(test_name, login, password) as started_session:
         yield started_session
+
+
+@fixture(scope="module")
+def activation_key(module_org):
+    ak = nailgun.entities.ActivationKey(
+        auto_attach=True,
+        content_view=module_org.default_content_view.id,
+        environment=module_org.library.id,
+        name=gen_string("alpha"),
+        organization=module_org,
+    ).create()
+    yield ak
+    ak.delete()
+
+
+@fixture(scope="module")
+def attach_subscription(module_org, activation_key):
+    for subs in nailgun.entities.Subscription(organization=module_org).search():
+        if subs.name == DEFAULT_SUBSCRIPTION_NAME:
+            # "quantity" must be 1, not subscription["quantity"]. Greater
+            # values produce this error: "RuntimeError: Error: Only pools
+            # with multi-entitlement product subscriptions can be added to
+            # the activation key with a quantity greater than one."
+            activation_key.add_subscriptions(data={"quantity": 1, "subscription_id": subs.id})
+            break
+    else:
+        raise Exception(f"{module_org.name} organization doesn't have any subscription")
```

Comment:
I think the message of the exception is not quite right. That line can be reached when the  `if` on line 121 is never satisfied. That doesn't necessary mean the org does not have any subscription.

Reply:
Old exception message changed.